var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -127,24 +132,47 @@ def cmake_args(self):
             self.define_from_variant('HIOP_TEST_WITH_BSUB', 'jsrun'),
         ])
 
+        # NOTE: If building with spack develop on a cluster, you may want to
+        # change the ctest launch command to use your job scheduler like so:
+        #
+        # args.append(
+        #     self.define('HIOP_CTEST_LAUNCH_COMMAND', 'srun -t 10:00'))
+
         if '+mpi' in spec:
             args.extend([
                 self.define('MPI_HOME', spec['mpi'].prefix),
                 self.define('MPI_C_COMPILER', spec['mpi'].mpicc),
                 self.define('MPI_CXX_COMPILER', spec['mpi'].mpicxx),
                 self.define('MPI_Fortran_COMPILER', spec['mpi'].mpifc),
             ])
+            # NOTE: On Cray platforms, libfabric is occasionally not picked up
+            # by Spack, causing HiOp's CMake code to fail to find MPI Fortran
+            # libraries. If this is the case, adding the following lines may
+            # resolve the issue. Searching <builddir>/CMakeFiles/CMakeError.log
+            # for MPI Fortran errors is the fastest way to check for this error.
+            #
+            # args.append(
+            #     self.define('MPI_Fortran_LINK_FLAGS',
+            #         '-L/path/to/libfabric/lib64/ -lfabric'))
 
         if '+cuda' in spec:
             cuda_arch_list = spec.variants['cuda_arch'].value
-            cuda_arch = cuda_arch_list[0]
-            if cuda_arch != 'none':
-                args.extend([
-                    self.define('HIOP_NVCC_ARCH', 'sm_{0}'.format(cuda_arch)),
-                    self.define('CMAKE_CUDA_ARCHITECTURES', cuda_arch),
-                ])
-            if '+magma' in spec:
-                args.append(self.define('HIOP_MAGMA_DIR', spec['magma'].prefix))
+            if cuda_arch_list[0] != 'none':
+                args.append(self.define('CMAKE_CUDA_ARCHITECTURES', cuda_arch_list))
+
+        # NOTE: if +rocm, some HIP CMake variables may not be set correctly.
+        # Namely, HIP_CLANG_INCLUDE_PATH. If the configure phase fails due to
```

Comment:
(question) Do you think it might help for `ROCmPackage` to set this universally (this would be hard now because it is agnostic to the possible build system, but I could think about how to make that work), or would doing that cause problems in other cases?

Do you have a notion of when this needs to be set (e.g. on particular systems)?

Reply:
I believe this is an issue with the newer versions of CMake, which is afaict why RAJA and Umpire (for example) disallow newer CMake versions. I've run into this on several systems with ROCm, [and based on this issue in the HIP repo](https://github.com/ROCm-Developer-Tools/HIP/issues/2318), it seems to be an issue with CMake's assumptions about ROCm's install tree. Hoping this gets fixed in newer CMakes so we can just bump required CMake version and get rid of this note.